core/src/test/java/io/questdb/test/cutlass/pgwire/PGJobContextTest.java

```diff
@@ -1772,13 +1774,59 @@ public void testCairoException() throws Exception {
         });
     }
 
+    @Test
+    public void testCancelRunningCreateQuery() throws Exception {
+        String[] queries = {"create table new_tab as (select count(*) from tab t1 cross join tab t2 where t1.x > 0)",
+                "select count(*) from tab t1 cross join tab t2 where t1.x > 0",
+                "insert into dest select count(*)::timestamp, 0, 0.0 from tab t1 cross join tab t2 where t1.x > 0",
+                "update dest set l = 1 from tab t1 where " +
+                        "'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' || t1.x = " +
+                        "'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA100000' "
+        };
+
+        assertWithPgServer(CONN_AWARE_EXTENDED_BINARY, (connection, binary) -> {
+            compiler.compile("create table if not exists tab as (select x::timestamp ts, x, rnd_double() d from long_sequence(1000000)) timestamp(ts) partition by day", sqlExecutionContext);
+            compiler.compile("create table if not exists dest as (select 1::long l)", sqlExecutionContext);
+            mayDrainWalQueue();
+
+            for (String query : queries) {
+                AtomicBoolean isCancelled = new AtomicBoolean(false);
+                CountDownLatch finished = new CountDownLatch(1);
+
+                try (final PreparedStatement stmt = connection.prepareStatement(query)) {
+                    new Thread(() -> {
+                        PGConnection pgCon = (PGConnection) connection;
+                        try {
+                            while (!isCancelled.get()) {
+                                Os.sleep(1);
+                                pgCon.cancelQuery();
+                            }
+                        } catch (SQLException e) {
+                            throw new RuntimeException(e);
+                        } finally {
+                            finished.countDown();
+                        }
+                    }, "cancellation thread").start();
+                    try {
```

Comment:
Please try with Os.sleep() here to check that the test is stable.

Reply:
Added in fd45433e .

Reply:
I wanted you to try locally, without committing. Would be good if you remove the sleep now.

Reply:
Done in latest commit .